<launch>
  <arg name="model_name" default="wamv" />
  <arg name="uuv_name" default="$(arg model_name)"/>
  <arg name="joy_id" default="0"/>

  <include file="$(find wamv_thruster_manager)/launch/thruster_manager.launch" />

  <group ns="$(arg uuv_name)">
    <rosparam file="$(find cascaded_pid_controller)/config/$(arg model_name)/inertial.yaml" command="load"/>

    <node pkg="cascaded_pid_controller" type="AccelerationControl.py" name="acceleration_control" output="screen">
      <remap from="/wamv/pose_gt" to="/odometry/filtered"/>
    </node>
  </group>

  <include file="$(find wamv_teleop)/launch/wamv_teleop.launch">
    <arg name="uuv_name" value="$(arg uuv_name)"/>
    <arg name="joy_id" value="$(arg joy_id)"/>
    <arg name="output_topic" value="cmd_accel"/>
    <arg name="message_type" value="accel"/>
    <arg name="gain_roll" default="0.2"/>
    <arg name="gain_pitch" default="0.2"/>
  </include>

</launch>
